### PR TITLE
[lldb] Add the SwiftEnableBareSlashRegexOnPlayground setting

### PR DESCRIPTION
**Pull request by augusto2112**
rdar://92690194

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -8303,6 +8303,18 @@ bool SwiftASTContext::GetImplicitImports(
   return true;
 }
 
+void SwiftASTContext::LoadStringProcessingModule(TargetSP target,
+                                                 ProcessSP process) {
+  static std::once_flag once_flag;
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: I think the once flag need to be owned by SwiftASTContext and not global. lldb-rpc-server is very long-lived.

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 2]
```diff
@@ -8303,6 +8303,17 @@ bool SwiftASTContext::GetImplicitImports(
   return true;
 }
 
+void SwiftASTContext::LoadStringProcessingModule(TargetSP target,
+                                                 ProcessSP process) {
+  llvm::call_once(m_load_string_processing_lib_once, [&]() {
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: IIUC, the call_once is redundant?

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h [hunk 3]
```diff
@@ -760,6 +760,13 @@ class SwiftASTContext : public TypeSystemSwift {
           &modules,
       Status &error);
 
+  // FIXME: the correct thing to do would be to get the modules by calling
+  // CompilerInstance::getImplicitImportInfo, instead of explicitly loading this
+  // module. However, we currently don't have  access to a CompilerInstance,
+  // which is why this function is needed.
+  void LoadStringProcessingModule(lldb::TargetSP target,
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h]: Should this be called LoadImplicitImports and also include _Concurrency?

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp (hunk 2 above)
Comment (augusto2112) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: On further inspection I believe it's redundant as `LoadOneModule` checks if the module is cached before loading it.

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp [hunk 4]
```diff
@@ -1359,6 +1359,11 @@ static llvm::Expected<ParsedExpression> ParseAndImport(
   invocation.getFrontendOptions().ModuleName = expr_name_buf;
   invocation.getIRGenOptions().ModuleName = expr_name_buf;
 
+  invocation.getLangOptions().EnableBareSlashRegexLiterals =
+      sc.target_sp->GetSwiftEnableBareSlashRegex();
+  invocation.getLangOptions().EnableExperimentalStringProcessing =
+      sc.target_sp->GetSwiftEnableBareSlashRegex();
```
Comment (adrian-prantl) [inline lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1365]: Nit: this is susceptible to a race condition if the setting gets changed between these two calls. Would be nice to store it in a bool first.

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h [hunk 5]
```diff
@@ -840,6 +840,7 @@ class SwiftASTContext : public TypeSystemSwift {
   std::unique_ptr<swift::irgen::IRGenerator> m_ir_generator_ap;
   std::unique_ptr<swift::irgen::IRGenModule> m_ir_gen_module_ap;
   llvm::once_flag m_ir_gen_module_once;
+  llvm::once_flag m_load_string_processing_lib_once;
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h:843]: Delete this?

**Review by adrian-prantl — APPROVED** [review]
LGTM with two tiny comments addressed. Thanks!

**Comment by adrian-prantl** [comment]
Oh.. and shouldn't we be able to write a testcase for this, maybe by using a regex in an expression?

**Pull request merged**